source/Politics.cpp

```diff
@@ -333,13 +333,15 @@ double Politics::Reputation(const Government *gov) const
 
 void Politics::AddReputation(const Government *gov, double value)
 {
-	reputationWith[gov] += value;
+	SetReputation(gov, reputationWith[gov] + value);
 }
 
 
 
 void Politics::SetReputation(const Government *gov, double value)
 {
+	value = min(value, gov->ReputationMax());
+	value = max(value, gov->ReputationMin());
```

Comment:
```suggestion
	value = max(min(value, gov->ReputationMax()), gov->ReputationMin());
```
I think its the same, and isnt this better because we only give a value to it once?

Reply:
It does do the same thing, but the compiler will probably produce the same machine code for both, and the current way is far more readable imo.
(Even when compiling with no optimisation, the difference is your way does 1 less `mov` operation, out of a dozen, and one less `lea`, out of 3, with a bunch of other stuff going on, too. The compiler optimised version is far shorter than either unoptimised version so I'd take the readability over the tiny performance gain when the compiler isn't optimising.)